route/route.go

```diff
@@ -32,67 +32,79 @@ func WithParam(ctx context.Context, p, v string) context.Context {
 	return context.WithValue(ctx, param(p), v)
 }
 
-// handle turns a Handle into httprouter.Handle
-func handle(h http.HandlerFunc) httprouter.Handle {
-	return func(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
-		ctx, cancel := context.WithCancel(context.Background())
+type contextFn func(r *http.Request) context.Context
+
+// Router wraps httprouter.Router and adds support for prefixed sub-routers
+// and per-request context injections.
+type Router struct {
+	rtr    *httprouter.Router
+	prefix string
+	ctxFn  contextFn
+}
+
+// New returns a new Router.
+func New(ctxFn contextFn) *Router {
```

Comment:
Yes, I need a context function exactly for the Frankenstein case: to pull out per-request multi-tenancy information from the HTTP request headers and stuff that user info into the context.

Or did I misunderstand your question?